.github/workflows/docs-edit-automation.yml

```
@@ -0,0 +1,180 @@
+name: Documentation edit automation
+
+on:
+  push:
+    branches:
+      - 'docs-enhancement/**'
+      - 'docs-bugfix/**'
+
+permissions:
+  contents: write
+  pull-requests: write
+
+jobs:
+  auto-update-docs:
+    # Skip if pushed by bot to avoid loops
+    if: github.actor != 'github-actions[bot]'
+    runs-on: ubuntu-latest
+    
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Setup Go
+        uses: actions/setup-go@v5
+        with:
+          go-version: '1.24'
```

See https://github.com/actions/setup-go#getting-go-version-from-the-gomod-file


```suggestion
          go-version-file: 'go.mod'
```